[BUG] Lock file is already being held

```
npm ERR! code 1
npm ERR! path /Users/pfeldman/code/playwright/packages/playwright-webkit
npm ERR! command failed
npm ERR! command sh -c node install.js
npm ERR! /Users/pfeldman/code/playwright/node_modules/proper-lockfile/lib/lockfile.js:68
npm ERR!                 return callback(Object.assign(new Error('Lock file is already being held'), { code: 'ELOCKED', file }));
npm ERR!                                               ^
npm ERR! 
npm ERR! Error: Lock file is already being held
npm ERR!     at /Users/pfeldman/code/playwright/node_modules/proper-lockfile/lib/lockfile.js:68:47
npm ERR!     at callback (/Users/pfeldman/code/playwright/node_modules/graceful-fs/polyfills.js:299:20)
npm ERR!     at FSReqCallback.oncomplete (node:fs:199:5) {
npm ERR!   code: 'ELOCKED',
npm ERR!   file: '/Users/pfeldman/Library/Caches/ms-playwright'
npm ERR! }

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/pfeldman/.npm/_logs/2021-10-17T18_02_49_591Z-debug.log
```

--- a/packages/playwright-core/src/utils/registry.ts
+++ b/packages/playwright-core/src/utils/registry.ts
@@ -524,22 +524,23 @@ export class Registry {
     const executables = this._addRequirementsAndDedupe(executablesToInstall);
     await fs.promises.mkdir(registryDirectory, { recursive: true });
     const lockfilePath = path.join(registryDirectory, '__dirlock');
-    const releaseLock = await lockfile.lock(registryDirectory, {
-      retries: {
-        retries: 10,
-        // Retry 20 times during 10 minutes with
-        // exponential back-off.
-        // See documentation at: https://www.npmjs.com/package/retry#retrytimeoutsoptions
-        factor: 1.27579,
-      },
-      onCompromised: (err: Error) => {
-        throw new Error(`${err.message} Path: ${lockfilePath}`);
-      },
-      lockfilePath,
-    });
     const linksDir = path.join(registryDirectory, '.links');
 
+    let releaseLock;
     try {
+      releaseLock = await lockfile.lock(registryDirectory, {
+        retries: {
+          retries: 10,
+          // Retry 20 times during 10 minutes with
+          // exponential back-off.
+          // See documentation at: https://www.npmjs.com/package/retry#retrytimeoutsoptions
+          factor: 1.27579,
+        },
+        onCompromised: (err: Error) => {
+          throw new Error(`${err.message} Path: ${lockfilePath}`);
+        },
+        lockfilePath,
+      });
       // Create a link first, so that cache validation does not remove our own browsers.
       await fs.promises.mkdir(linksDir, { recursive: true });
       await fs.promises.writeFile(path.join(linksDir, calculateSha1(PACKAGE_PATH)), PACKAGE_PATH);
@@ -574,7 +575,8 @@ export class Registry {
         throw e;
       }
     } finally {
-      await releaseLock();
+      if (releaseLock)
+        await releaseLock();
     }
   }
 

